orca-uv.py

```diff
@@ -0,0 +1,417 @@
+# -*- coding: utf-8 -*-
+'''
+
+# orca-uv
+
+'''
+
+import sys  # sys files processing
+import os  # os file processing
+import re  # regular expressions
+import argparse  # argument parser
+import numpy as np  # summation
+from mpi4py import MPI
+import math
+import matplotlib.pyplot as plt  # plots
+from scipy.signal import find_peaks  # peak detection
+
+from utils import nsplit, gauss, read_orca_output, draw_2Dmols
+
+plt.rcParams.update({"font.size": 22})
+
+# global constants
+found_uv_section = False  # check for uv data in out
+specstring_start = 'ABSORPTION SPECTRUM VIA TRANSITION ELECTRIC DIPOLE MOMENTS'  # check orca.out from here
+
+ORCA_METHOD = "EOM-CCSD"
+
+if ORCA_METHOD == "TD-DFT":
+    specstring_end = 'ABSORPTION SPECTRUM VIA TRANSITION VELOCITY DIPOLE MOMENTS'  # stop reading orca.out from here
+elif ORCA_METHOD == "EOM-CCSD":
+    specstring_end = "CD SPECTRUM" # stop reading orca.out from here
+
+w_wn = 1000  # w = line width for broadening - wave numbers, FWHM
+w_nm = 10  # w = line width for broadening - nm, FWHM
+export_delim = " "  # delimiter for data export
+
+# plot config section - configure here
+nm_plot = True  # wavelength plot /nm if True, if False wave number plot /cm-1
+show_single_gauss = True  # show single gauss functions if True
+show_single_gauss_area = True  # show single gauss functions - area plot if True
+show_conv_spectrum = True  # show the convoluted spectra if True (if False peak labels will not be shown)
+show_sticks = True  # show the stick spectra if True
+label_peaks = True  # show peak labels if True
+minor_ticks = True  # show minor ticks if True
+show_grid = False  # show grid if True
+linear_locator = True  # tick locations at the beginning and end of the spectrum x-axis, evenly spaced
+spectrum_title = "Absorption spectrum"  # title
+spectrum_title_weight = "bold"  # weight of the title font: 'normal' | 'bold' | 'heavy' | 'light' | 'ultrabold' | 'ultralight'
+y_label = "intensity"  # label of y-axis
+x_label_wn = r'energy /cm$^{-1}$'  # label of the x-axis - wave number
+x_label_nm = r'$\lambda$ /nm'  # label of the x-axis - nm
+plt_y_lim = 0.4
+figure_dpi = 300  # DPI of the picture
+
+# global lists
+statelist = list()  # mode
+energylist = list()  # energy cm-1
+intenslist = list()  # fosc
+gauss_sum = list()  # list for the sum of single gaussian spectra = the convoluted spectrum for cm-1
+
+
+# plot config section - configure here
+nm_plot = True  # wavelength plot in nm if True, if False energy plot in eV
```

Comment:
Next few lines are repeated variable declarations of lines 38-53.

Reply:
Thanks for catching this.